File: C++/Zad6.cpp
// Zadanie numer 6 Lista 1
// Liczenie ilosci samoglosek i zapisywanie do pliku z parametru

#include <iostream>
#include <string>
#include <fstream>

using namespace std;


int main(int argc, char* argv[])
{
	if (argc != 2) cout << "Podano nieprawidlowa liczba parametrow";
	else
	{
    setlocale(LC_ALL, "");
    int ilosc, licznik = 0;
    bool flaga_spacja = 0;
    string tekst;

    cout << "Podaj tekst: " << endl;
    getline(cin, tekst);
    string::size_type rozmiar = tekst.size();
    ilosc = static_cast<int>(rozmiar);

    for (int i = 0; i < ilosc; i++)
    {
        if (tekst[i] == 'a' || tekst[i] == '¹' || tekst[i] == 'y' || tekst[i] == 'i'
            || tekst[i] == 'e' || tekst[i] == 'ê' || tekst[i] == 'u' || tekst[i] == 'O'
            || tekst[i] == 'A' || tekst[i] == '¥' || tekst[i] == 'Y' || tekst[i] == 'I'
            || tekst[i] == 'E' || tekst[i] == 'Ê' || tekst[i] == 'U' || tekst[i] == 'o')

            licznik++;
            
        if (tekst[i] == ' ') flaga_spacja=1;
    }
    if (flaga_spacja==1) cout << "Blad - nalezy wprowadzic tylko jeden wyraz";
    else
    {
    fstream plik;
    plik.open(argv[1], ios::out | ios::app);
    plik << tekst << "\t" << licznik << endl;
    plik.close();
    return 0;
    }
    }
}

File: Python/Zad6.py
import sys
if(len(sys.argv) !=2):
    print('Podano nieprawidlowa ilosc argumentow')
else:
    try:
        nazwa_pliku = sys.argv[1] 
        slowo_wejsciowe = input("Wprowadź słowo: ")
        spolgloski="b, c, ć, d, f, g, h, j, k, l, ł, m, n, ń, p, r, s, ś, t, w, z, ż, ź, B, C, Ć, D, F, G, H, J, K, L, M, N, P, R, S, T, W, Z, Ź, Ż, Ł"
        plik = open(nazwa_pliku, 'a')
        ilosc_spolglosek = 0
        for znak in slowo_wejsciowe: 
            if(spolgloski.__contains__(znak)):
                ilosc_spolglosek += 1 
        plik.write(slowo_wejsciowe + ';' + str(ilosc_spolglosek) + '\n')
        print (slowo_wejsciowe + ';' + str(ilosc_spolglosek))
        plik.close()
    except ValueError:
        print('Nie podano prawidlowych argumentow')
    
